### PR TITLE
Add must_use annotations to all types that implerment Futures/Streams

### PR DESCRIPTION
**Pull request by dvdplm**
Closes https://github.com/libp2p/rust-libp2p/issues/333

### CODE REVIEW

**Comment by dvdplm** [comment]
I don't have a strong opinion on this. I guess that one persons "it's totally obvious" might be non-obvious to another, but yeah, either way is fine with me.

File: core/src/nodes/handled_node_tasks.rs [hunk 1]
```diff
@@ -51,6 +51,7 @@ use PeerId;
 
 /// Implementation of `Stream` that handles a collection of nodes.
 // TODO: implement Debug
+#[must_use = "streams do nothing unless polled"]
```
Comment (twittner) [inline core/src/nodes/handled_node_tasks.rs]: `HandledNodesTasks` is not a stream.

File: core/src/nodes/listeners.rs [hunk 2]
```diff
@@ -74,6 +74,7 @@ use {Multiaddr, Transport};
 /// tokio::run(future.map_err(|_| ()));
 /// # }
 /// ```
+#[must_use = "streams do nothing unless polled"]
```
Comment (twittner) [inline core/src/nodes/listeners.rs]: As `ListenersStream` does not implement `Stream` it is better to refer to it directly, as in "ListenersStream does nothing unless polled".

Comment (twittner) [reply core/src/nodes/listeners.rs]: I stand corrected. It actually implements `Stream`.

**Comment by tomaka** [comment]
Well, I'm going to close as basically every single line of this PR conflicts with master.

**Pull request closed without merging**